workspaces/cli-shared/src/captures/avro/file-system/capture-saver.ts

```diff
@@ -101,13 +109,39 @@ export class CaptureSaver implements ICaptureSaver {
   }
 
   async save(sample: IHttpInteraction) {
+    this.interactionsReceivedCount++;
     // don't await flush, just enqueue
-    await this.batcher.add(sample);
+    this.batcher.add(sample);
```

Comment:
batcher.add() returns a Promise that resolves once the request has been flushed to a "batch" event. Per https://github.com/SGrondin/bottleneck/blob/master/src/Batcher.coffee it resolves after every "batch" event handler has been called _synchronously_, so it doesn't wait for the async work we are doing.

Reply:
Might be worth doing something with the rejection of the Promise  that `this.batcher.add` returns. Without us redirecting it, it'll be a cryptic uncaught rejection from `Batcher` that has lost trace of this `save` method. For these kinds of errors that can happen out of band emitting an `error` event in a `EventEmitter` the `CaptureSaver` is generally the way to go. If a consumer of `CaptureSaver` doesn't handle it, it'll cause a crash, which is exactly what you want unhandled errors to do.